### PR TITLE
[wasm] Move hardcoded `_BoolPropertiesThatTriggerRelinking` to `wasm-props.json` included in the runtime pack

### LINKED ISSUE #91628: [wasm] Move the default property settings that were used to build the runtime pack, to `wasm-props.json` included in the runtime pack
This should replace the hardcoded table in `WasmApp.Native.targets` - `_BoolPropertiesThatTriggerRelinking`.

cc @pavelsavara

### PR DESCRIPTION
**Pull request by ilonatommy**
Fixes https://github.com/dotnet/runtime/issues/91628.

Wasi wbt had an error of not setting `WasmBuildNative=true` in  `NativeBuildOrPublishFails` - this PR fixes it.

### CODE REVIEW

File: src/mono/browser/browser.proj [hunk 1]
```diff
@@ -332,8 +332,8 @@
       { "identity": "WasmNativeStrip",              "defaultValueInRuntimePack": "true" },
       { "identity": "WasmSingleFileBundle",         "defaultValueInRuntimePack": "false" },
       { "identity": "WasmEnableSIMD",               "defaultValueInRuntimePack": "true" },
-      { "identity": "WasmEnableExceptionHandling",  "defaultValueInRuntimePack": "true" },
-    ],
+      { "identity": "WasmEnableExceptionHandling",  "defaultValueInRuntimePack": "true" }
```
Comment (maraf) [inline src/mono/browser/browser.proj:335]: Could we generate the JSON from actual property values?

Comment (ilonatommy) [reply src/mono/browser/browser.proj:335]: Why do we need actual property values? How do we extract runtime pack actual property values?
- In this place only `WasmEnableExceptionHandling` and `WasmEnableSIMD` are populated.
- Do the defaults change in time?

**Review by maraf — APPROVED** [review]

**Comment by ilonatommy** [comment]
Failures are https://github.com/dotnet/runtime/issues/46020

**Comment by ilonatommy** [comment]
/ba-g failures are known and happening randomly

**Comment by ilonatommy** [comment]
/ba-g infrastructure issues - timeouts on non-related lanes

**Pull request merged**